apps/jobs/lib/jobs/declaration_termination_job.ex

```diff
@@ -0,0 +1,130 @@
+defmodule Jobs.DeclarationTerminationJob do
+  @moduledoc false
+
+  import Core.Registers.API,
+    only: [create_register_entry: 1, update_register: 2, prepare_register_update_data: 4]
+
+  import Core.Registers.Validator
+
+  alias Core.Registers.Register
+  alias Core.Registers.RegisterEntry
+  alias Jobs.Jabba.Client, as: JabbaClient
+  alias Jobs.Jabba.Task, as: JabbaTask
+
+  @declaration_terminate_task_type JabbaTask.type(:declaration_terminate)
+  @declaration_termination_job_type JabbaClient.type(:declaration_termination)
+
+  @status_matched RegisterEntry.status(:matched)
+  @status_not_found RegisterEntry.status(:not_found)
+  @status_error RegisterEntry.status(:error)
+  @status_processed RegisterEntry.status(:processed)
+
+  @ops_api Application.get_env(:core, :api_resolvers)[:ops]
+
+  def search_jobs(filter, order_by, limit, offset) do
+    filter
+    |> Kernel.++([{:type, :equal, @declaration_termination_job_type}])
+    |> JabbaClient.search_jobs(order_by, {offset, limit})
+  end
+
+  def get_job(id) do
+    case JabbaClient.get_job(id) do
+      {:ok, job} -> {:ok, job}
+      nil -> {:ok, nil}
+    end
+  end
+
+  def create(processed_data) do
+    tasks = get_terminate_declaration_tasks(processed_data)
+    JabbaClient.create_job(tasks, @declaration_termination_job_type, name: "Terminate declaration")
```

Comment:
```suggestion
    opts = [
      name: "Terminate declaration",
      callback: {"ehealth", Jobs.DeclarationTerminationJob, :job_processed, []},
      strategy: "CONCURRENT"
    ]  
      JabbaClient .create_job(tasks, @declaration_termination_job_type, opts)
```
Callback will called when job processed or failed.
So you don't need indexes in tasks anymore.